samples/extensions/extended_dynamic_state2/README.md

```diff
@@ -0,0 +1,259 @@
+	    VK_PRIMITIVE_TOPOLOGY_TRIANGLE_LIST, /* value used in 1st and 2nd pipeline */
+	    0,
+	    VK_FALSE); /* primitiveRestartEnable */
+
+VkPipelineRasterizationStateCreateInfo rasterization_state =
+	vkb::initializers::pipeline_rasterization_state_create_info(
+	    VK_POLYGON_MODE_FILL, /* value used in 1st, 2nd and 3rd pipeline */
+	    VK_CULL_MODE_BACK_BIT,
+	    VK_FRONT_FACE_CLOCKWISE,
+	    0);
+rasterization_state.depthBiasConstantFactor = 1.0f;
+rasterization_state.depthBiasSlopeFactor    = 1.0f;
+rasterization_state.depthBiasClamp          = 0.0f;
+
+/* Note: Using Reversed depth-buffer for increased precision, so greater depth values are kept */
+VkPipelineDepthStencilStateCreateInfo depth_stencil_state =
+	vkb::initializers::pipeline_depth_stencil_state_create_info(
+	    VK_TRUE, /* depthTestEnable */
+	    VK_TRUE, /* depthWriteEnable */
+	    VK_COMPARE_OP_GREATER);
+...
+
+/* VkGraphicsPipelineCreateInfo for all pipelines, parameters are modified before each pipeline creation */
+VkGraphicsPipelineCreateInfo graphics_create{VK_STRUCTURE_TYPE_GRAPHICS_PIPELINE_CREATE_INFO};
+graphics_create.pInputAssemblyState = &input_assembly_state;
+graphics_create.pRasterizationState = &rasterization_state;
+graphics_create.pDepthStencilState  = &depth_stencil_state;
+graphics_create.pTessellationState  = VK_NULL_HANDLE;
+...
+
+VK_CHECK(vkCreateGraphicsPipelines(get_device().get_handle(), pipeline_cache, 1, &graphics_create, VK_NULL_HANDLE, &pipeline1));
+
+/* Second pipeline creation */
+rasterization_state.rasterizerDiscardEnable = VK_TRUE;
+
+VK_CHECK(vkCreateGraphicsPipelines(get_device().get_handle(), pipeline_cache, 1, &graphics_create, VK_NULL_HANDLE, &pipeline2));
+
+/* Third pipeline creation */
+rasterization_state.rasterizerDiscardEnable = VK_FALSE;
+input_assembly_state.topology = VK_PRIMITIVE_TOPOLOGY_TRIANGLE_LIST;
+input_assembly_state.primitiveRestartEnable = VK_TRUE;
+
+VK_CHECK(vkCreateGraphicsPipelines(get_device().get_handle(), pipeline_cache, 1, &graphics_create, VK_NULL_HANDLE, &pipeline3));
+
+/* Fourth pipeline creation */
+VkPipelineTessellationStateCreateInfo tessellation_state = 	vkb::initializers::pipeline_tessellation_state_create_info(3);
+graphics_create.layout = pipeline_layouts.tesselation;
+graphics_create.pTessellationState  = &tessellation_state;
+input_assembly_state.topology = VK_PRIMITIVE_TOPOLOGY_PATCH_LIST;
+input_assembly_state.primitiveRestartEnable = VK_FALSE;
+rasterization_state.depthBiasEnable = VK_TRUE;
+if (get_device().get_gpu().get_features().fillModeNonSolid)
+{
+	rasterization_state.polygonMode = VK_POLYGON_MODE_LINE; /* Wireframe mode */
+}
+    
+VK_CHECK(vkCreateGraphicsPipelines(get_device().get_handle(), pipeline_cache, 1, &graphics_create, VK_NULL_HANDLE, &pipeline4));
+```
+
+In the above approach if developer would like to change the patch control points number, then for each different number a new pipeline would be required.
```

Comment:
We still have issue with presenting dynamic change of the patch control points number.
When we use triangle based model (together with suitable tessellation shaders) change of the  patch control points number is _breaking_ the model.
We had idea to use quad model, but it would require new pipeline with new shaders, so again we would not show the dynamic change of the patch control points number.